python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -330,6 +331,18 @@ def _report_current_usage(
             f"{cur_usage.overall.object_store_memory_str()}/"
             f"{limits.object_store_memory_str()} object_store_memory"
         )
+        if (
+            cur_usage.overall.object_store_memory is not None
+            and limits.object_store_memory is not None
+            and cur_usage.overall.object_store_memory > limits.object_store_memory
+        ):
+            spilled_obj_memory = (
+                cur_usage.overall.object_store_memory - limits.object_store_memory
+            )
```

Comment:
This isn't accurate. Because 1) `limits.object_store_memory` isn't the actual object store memory, we use 25% of that by default; 2) due to data skew, object store usage could be imbalanced. 
So unfortunately, it's not easy to know the actual object spilling, without querying all object stores.

Reply:
also, we've already printed both values above. this is redundant?

Reply:
I was assuming that the difference of them is the spilled object memory.
If that is not the case, is there a way to query all object stores? 
If there is a way, does the query time worthy for the information?

Reply:
> also, we've already printed both values above. this is redundant?

I think there are different values and since there are no method to compute the difference of object memory, I have to compute it.

But I agree that it is easy to calculate the difference after seeing both values above.

Reply:
> This isn't accurate. Because 1) `limits.object_store_memory` isn't the actual object store memory, we use 25% of that by default; 2) due to data skew, object store usage could be imbalanced. So unfortunately, it's not easy to know the actual object spilling, without querying all object stores.

From my understanding, `limits.object_store_memory` specify the cluster's current limit, which doesn't need to be the actual object store memory.
From [here](https://docs.ray.io/en/latest/ray-core/scheduling/memory-management.html#concepts) it says that
> objects are [spilled to disk](https://docs.ray.io/en/latest/ray-core/objects/object-spilling.html#id1) if the object store fills up

which I think it means filling up the current limit.
And the difference can show how many bytes are spilled to the disk based on the current limit setting.

As for 2), I currently do not have enough knowledge to know why it can affect this.

Please feel free to correct me if I have any mis-understanding. Thanks!